src/utils/treeUtils.tsx

```diff
@@ -38,3 +38,26 @@ export const findEventParentByIri = (childIri: string, root: FaultEvent): FaultE
 
   return undefined;
 };
+
+export const findNodeByIri = (node, targetIri) => {
+  // If the current node matches the target IRI, return it
+  if (node.iri === targetIri) {
+    return [node];
+  }
+
+  // If the current node has children and it is an array, recursively search each child
+  if (Array.isArray(node.children)) {
+    for (const childNode of node.children) {
+      // Recursively search for the target IRI in the child node
+      const path = findNodeByIri(childNode, targetIri);
+
+      // If the target IRI is found in the child node, prepend the current node to the path and return it
+      if (path.length > 0) {
+        return [node, ...path];
+      }
+    }
+  }
+
+  // If the target IRI is not found in the current node or its children, return an empty array
```

Comment:
No need for comments, the name of the function is explicit enough

Reply:
Removed.